### PR TITLE
Integrating hipchat notifications

### PR DESCRIPTION
**Pull request by meatballhat**
that currently live in standalone tasks distributed via copypasta

### CODE REVIEW

**Comment by jszwedko** [comment]
:clap: :ice_cream: :shell: :zap:

File: library/modcloth_app_facts [hunk 1]
```diff
@@ -0,0 +1 @@
+modcloth_app_facts.py
```
Comment (rafecolton) [inline library/modcloth_app_facts:1]: Why the symlink in favor of naming the file w/o the `.py`?

File: tasks/main.yml [hunk 2]
```diff
@@ -1,4 +1,35 @@
...
+    repo={{ modcloth_app_github_repo }}
+    revision={{ modcloth_app_docker_tag }}
+  changed_when: False
+
+- name: render hipchat deployment html
+  sudo: false
+  connection: local
+  template:
+    src=hipchat_notification.frag.html.j2
+    dest={{ modcloth_app_hipchat_html }}
```
Comment (rafecolton) [inline tasks/main.yml:16]: This doesn't appear to be defaulted anywhere - is there a recommended default value?

File: library/modcloth_app_facts (hunk 1 above)
Comment (meatballhat) [inline library/modcloth_app_facts:1]: File extensions on modules are the new hotness upstream, although the distribution process appears to copy them and remove the extension.  This makes it play nicely with github syntax highlighting, for one, so I'm in favor of the move.

Comment (rafecolton) [inline library/modcloth_app_facts:1]: :thumbsup:

File: tasks/main.yml (hunk 2 above)
Comment (meatballhat) [inline tasks/main.yml:16]: Are you referring to `modcloth_app_hipchat_html`?  If so, it's covered by the `modcloth_app_facts` module.

File: library/modcloth_app_facts.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    module = AnsibleModule(
+        argument_spec=dict(
+            github_token=dict(required=True),
+            repo=dict(required=True),
+            revision=dict(required=True),
+        )
+    )
+    module.exit_json(
+        changed=False,
+        ansible_facts=dict(
```
Comment (rafecolton) [inline library/modcloth_app_facts.py:24]: I assume this sets the facts in the external context?  If so, that's pretty slick.  Did you happen to find documentation anywhere for what params may be passed to `exit_json` or is it just in the source?

File: tasks/main.yml (hunk 2 above)
Comment (rafecolton) [inline tasks/main.yml:16]: Never mind, I see it's being set by the module :smile:

**Comment by rafecolton** [comment]
This is awesome, thank you!!!

File: library/modcloth_app_facts.py (hunk 3 above)
Comment (meatballhat) [inline library/modcloth_app_facts.py:24]: `exit_json` accepts arbitrary key-value stuff, and here's the bit about `ansible_facts`: http://docs.ansible.com/developing_modules.html#module-provided-facts

Comment (meatballhat) [inline library/modcloth_app_facts.py:24]: The bit about `exit_json` is buried in this section: http://docs.ansible.com/developing_modules.html#common-module-boilerplate

File: library/modcloth_app_facts (hunk 1 above)
Comment (jszwedko) [inline library/modcloth_app_facts:1]: I'm seeing behaviour where it doesn't copy the file (nor does it copy down the symlink) when install via `ansible-galaxy`. This is with ansible 1.7.2.

**Pull request merged**